cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartDownscaleStopInstancesHandler.java

```diff
@@ -0,0 +1,90 @@
+package com.sequenceiq.cloudbreak.cloud.handler;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudInstance;
+import com.sequenceiq.cloudbreak.cloud.model.CloudVmInstanceStatus;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleStopInstancesRequest;
+import com.sequenceiq.cloudbreak.cloud.event.instance.StopStartDownscaleStopInstancesResult;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopStartDownscaleStopInstancesHandler implements CloudPlatformEventHandler<StopStartDownscaleStopInstancesRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StopStartDownscaleStopInstancesHandler.class);
+
+    private static final Long STOP_POLL_TIMEBOUND_MS = 300_000L;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public Class<StopStartDownscaleStopInstancesRequest> type() {
+        return StopStartDownscaleStopInstancesRequest.class;
+    }
+
+    @Override
+    public void accept(Event<StopStartDownscaleStopInstancesRequest> event) {
+        StopStartDownscaleStopInstancesRequest request = event.getData();
+        LOGGER.info("StopStartDownscaleStopInstancesHandler: {}", event.getData().getResourceId());
+
+        CloudContext cloudContext = request.getCloudContext();
+        try {
+            CloudConnector<?> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            List<CloudInstance> cloudInstancesToStop = request.getCloudInstancesToStop();
+            List<CloudVmInstanceStatus> cloudVmInstanceStatusList = Collections.emptyList();
+            if (cloudInstancesToStop.size() > 0) {
+                LOGGER.info("Attempting to stop instances with a timebound of {}ms. count={}, instances=[{}]",
+                        STOP_POLL_TIMEBOUND_MS, cloudInstancesToStop.size(), cloudInstancesToStop.stream().map(CloudInstance::getInstanceId)
+                                .collect(Collectors.toList()));
+                // TODO CB-14929: Error handlnig. Currently - an error from the following command (e.g. NPE)
```

Comment:
because it is an interrupted flow and it should not happen. Flow is finalized by the flow engine becuase we have a remediation code piece for badly written flows (if we dont do this, then the flow would stuck forever)

Reply:
Relying on this default flow handling by the engine. As mentioned in a previous comment - this is working well. Hence did not consider this set of changes to be critical w.r.t the first release.

Reply:
it is working well because I implemented this feature some months ago to avoid stuck flows, but it is an **emergency code** to avoid AD-HOC SQLs. But don't do this because it left the cluster in an improper state and there will be no notification for the user. https://github.com/hortonworks/cloudbreak/blob/4699316f4d962df36dc42ab596b7906ef2bbe802/core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/CloudbreakFlowInformation.java#L93-L100
**Without proper exception handling in flows / flowhandlers we CAN'T merge this PR.**

Reply:
Replied further up about a separate PR to change the flows to introduce error states, etc. Also found an issue with some transitions ... which I'm assuming this code must've been handling.

Reply:
PR merged. Resolving.